packages/bytecode-utils/src/lib/bytecode.ts

```diff
@@ -0,0 +1,88 @@
+import { arrayify, hexlify } from '@ethersproject/bytes';
+import bs58 from 'bs58';
+import * as CBOR from 'cbor-x';
+import { EthereumProvider } from 'ethereum-provider';
+
+type CBOR = {
+  bytes: string;
+  length: number;
+};
+
+// eslint-disable-next-line functional/no-mixed-type
+type DecodedObject = {
+  cbor: CBOR;
+  ipfs?: string;
+  solcVersion?: string;
+  [key: string]: string | CBOR | Uint8Array | undefined | boolean;
+};
+
+/**
+ * Decode contract's bytecode
+ * @param bytecode - hex of the bytecode
+ * @returns Object describing the contract
+ */
+export const decode = (bytecode: string): DecodedObject => {
+  if (bytecode.length === 0) {
+    throw Error('Bytecode cannot be null');
+  }
+  // Take latest 2 bytes of the bytecode (length of the cbor object)
+  const cborLength = parseInt(`${bytecode.slice(-4)}`, 16);
+
+  // Extract the cbor object using the extracted lenght
+  const cborRaw = bytecode.substring(
+    bytecode.length - 4 - cborLength * 2,
+    bytecode.length - 4
+  );
+
+  // cbor decode the object and get a json
+  const cborDecodedObject = CBOR.decode(arrayify(`0x${cborRaw}`));
+
+  const result: DecodedObject = {
+    cbor: {
+      bytes: `0x${cborRaw}`,
+      length: cborLength,
+    },
+  };
+
+  // Decode all the parameters from the json
+  Object.keys(cborDecodedObject).forEach((key: string) => {
+    switch (key) {
+      case 'ipfs': {
+        const ipfsCID = bs58.encode(cborDecodedObject.ipfs);
+        result.ipfs = ipfsCID;
+        break;
+      }
+      case 'solc': {
+        result.solcVersion = cborDecodedObject.solc.join('.');
+        break;
+      }
+      case 'experimental': {
+        result.experimental = cborDecodedObject.experimental;
+        break;
+      }
+      default: {
+        result[key] = hexlify(cborDecodedObject[key]);
+        break;
+      }
+    }
+  });
+
+  return result;
+};
+
+/**
+ * Get contract's bytecode given an address and an rpc
+ * @param address - contract's address
+ * @param provider - provider used to get the bytecode
+ * @returns bytecode
+ */
+export const get = async (
+  address: string,
+  provider: EthereumProvider
+): Promise<string> => {
+  const bytecode = await provider.request({
+    method: 'eth_getCode',
+    params: [address, 'latest'],
+  });
+  return bytecode as string;
+};
```

Comment:
Hmm do we really need this? I think this is not the job of this library. And I couldn't find the source code of the dependency https://www.npmjs.com/package/ethereum-provider

Reply:
That is just a type, you need to pass a provider when using it. I think it makes sense as long as we don't force specific dependecy to be used (web3, ethers, ...). This works in every environemnt because all the libraries have an Ethereum Provider, it's an EIP I believe.

Reply:
Where would this be needed? I still don't see any use case or need for this.

Reply:
Here for example: https://github.com/marcocastignoli/contract-call-decoder/blob/6304b2db3d88f80747e74e53d472c15a2e6085a2/src/lib/ContractCallDecoder.ts#L46

Reply:
I still think this is not this package's job and if they need a functioning Provider anyway, they'll likely use the wrapper function e.g. `web3.eth.getCode` anyway. But ok if you think it makes sense.

Reply:
Now I have doubts :S We have to understand if it makes sense to offer a shortcut to fetch the ABI, or we should push developers to use theirs library's option

Reply:
When in doubt, I usually opt in for not adding things but adding them when the need arises.

Reply:
removed